### PR TITLE
MGOMIRROR-466: ConvertLegacyIndexOptionsFromOp only removes one legac…

### PR DESCRIPTION
**Pull request by huan-Mongo**
…y index option

### CODE REVIEW

**Review by tfogo — CHANGES_REQUESTED** [review]
I think there's a typo in one of test cases, looks good once that's fixed.

File: common/bsonutil/indexes_test.go [hunk 1]
```diff
@@ -80,3 +81,27 @@ func TestConvertLegacyIndexKeys(t *testing.T) {
...
+		indexoptionsSingleInvalidOption := bson.D{{"v", int32(1)}, {"key", bson.D{{"a", int32(1)}}},
+			{"name", "a_1"}, {"unique", true}, {"invalid_option", int32(1)}}
+		ConvertLegacyIndexOptionsFromOp(&indexoptionsSingleInvalidOption)
+		So(indexoptionsSingleInvalidOption, ShouldResemble, convertedIndex)
+
+		indexoptionsMultipleInvalidOptions := bson.D{{"v", int32(1)}, {"key", bson.D{{"a", int32(1)}}},
+			{"name", "a_1"}, {"unique", true}, {"invalid_option_1", true}, {"invalid_option_2", int32(1)}}
+		ConvertLegacyIndexOptionsFromOp(&indexoptionsMultipleInvalidOptions)
+		So(indexoptionsMultipleInvalidOptions, ShouldResemble, convertedIndex)
+		fmt.Println(indexoptionsMultipleInvalidOptions)
```
Comment (edobranov) [inline common/bsonutil/indexes_test.go:105]: This `fmt.Println()` line might've been used for debugging; we should probably remove it from the test.

**Review by edobranov — CHANGES_REQUESTED** [review]

File: common/bsonutil/indexes_test.go [hunk 2]
```diff
@@ -80,3 +81,27 @@ func TestConvertLegacyIndexKeys(t *testing.T) {
...
 }
+
+func TestConvertLegacyIndexOptionsFromOp(t *testing.T) {
+	testtype.SkipUnlessTestType(t, testtype.UnitTestType)
+	convertedIndex := bson.D{{"v", int32(1)}, {"key", bson.D{{"a", int32(1)}}},
+		{"name", "a_1"}, {"unique", true}}
+
+	Convey("Converting legacy index options", t, func() {
+		indexoptionsNoInvalidOption := bson.D{{"v", int32(1)}, {"key", bson.D{{"a", int32(1)}}},
+			{"name", "a_1"}, {"unique", true}, {"invalid_option", int32(1)}}
```
[the remark this answers is not in the document]
Comment (huan-Mongo) [reply common/bsonutil/indexes_test.go]: Yes, you are right. I forgot to remove it after copying.

File: common/bsonutil/indexes_test.go (hunk 1 above)
Comment (huan-Mongo) [reply common/bsonutil/indexes_test.go]: Ah, right. Forgot to remove it.

**Review by edobranov — APPROVED** [review]
LGTM!

**Review by tfogo — APPROVED** [review]
LGTM!

**Pull request merged**